File: tutorias-inf-uva-api/src/main/java/es/uva/inf/tutorias/business/domain/models/Profesor.java
package es.uva.inf.tutorias.business.domain.models;

import java.util.List;

/**
 * The persistent class for the profesor database table.
 * 
 */
public class Profesor extends Usuario {
	private static final long serialVersionUID = 1L;
	
	private String centroHabitual;
	private String despacho;
	private String telefono;
	private List<HorarioTutorias> horariosTutorias;

	public Profesor() {
	}

	public String getCentroHabitual() {
		return centroHabitual;
	}

	public void setCentroHabitual(String centroHabitual) {
		this.centroHabitual = centroHabitual;
	}

	public String getDespacho() {
		return this.despacho;
	}

	public void setDespacho(String despacho) {
		this.despacho = despacho;
	}

	public String getTelefono() {
		return this.telefono;
	}

	public void setTelefono(String telefono) {
		this.telefono = telefono;
	}

	public List<HorarioTutorias> getHorariosTutorias() {
		return this.horariosTutorias;
	}

	public void setHorariosTutorias(List<HorarioTutorias> horariosTutorias) {
		this.horariosTutorias = horariosTutorias;
	}

}

File: tutorias-inf-uva-frontend/src/app/models/profesor.ts
import { HorarioTutorias } from "./horario-tutorias";
import { Usuario } from "./usuario";

export class Profesor extends Usuario {
	centroHabitual: string;
	despacho: string;
	telefono: string;
	horariosTutorias: HorarioTutorias[];

	constructor(profesorJSON: any) {
		super(profesorJSON);
		this.identificador = profesorJSON.identificador;
		this.nombre = profesorJSON.nombre;
		this.apellidos = profesorJSON.apellidos;
		this.nif = profesorJSON.nif;
		this.email = profesorJSON.email;
		this.titulaciones = profesorJSON.titulaciones;
		this.asignaturas = profesorJSON.asignaturas;
		this.centroHabitual = profesorJSON.centroHabitual;
		this.despacho = profesorJSON.despacho;
		this.telefono = profesorJSON.telefono;
		this.horariosTutorias = profesorJSON.horariosTutorias;
	}
	
}